drracket/drracket/private/syncheck/gui.rkt

```diff
@@ -2723,12 +2721,12 @@ If the namespace does not, they are colored the unbound color.
         (cond
           [(is-a? text syncheck-text<%>) text]
           [else 
-           (let ([admin (send text get-admin)])
-             (and (is-a? admin editor-snip-editor-admin<%>)
-                  (let* ([enclosing-editor-snip (send admin get-snip)]
-                         [editor-snip-admin (send enclosing-editor-snip get-admin)]
-                         [enclosing-editor (send editor-snip-admin get-editor)])
-                    (loop enclosing-editor))))])))
+           (define admin (send text get-admin))
+           (and (is-a? admin editor-snip-editor-admin<%>)
```

Comment:
Unpopular opinion: I'd rather convert `and` to `cond` here if that allows the `let*` conversion to `define`.

Reply:
I hold that same opinion, in general.

Reply:
That would look like this, right?

```
(cond
  [(not (is-a? admin editor-snip-editor-admin<%>)) #false]
  [else
   (define enclosing-editor-snip ...)
   (define editor-snip-admin ...)
   (define enclosing-editor ...)
   (loop enclosing-editor)])
```

I could be down with that. Are we thinking only in cases where there's two branches in the `and`, the first branch doesn't use `let`, and the second branch does?

Reply:
It occurs to me that it's also possible to rewrite this code like so:

```scheme
(and (is-a? admin editor-snip-editor-admin<%>)
     (loop (send+ admin (get-snip) (get-admin) (get-editor))))
```

Though I don't typically encounter `send+` in real code. Perhaps it's too esoteric?

Reply:
No strong opinion on the `send+` but I'd say that if any `and` has a `let` inside, I'd probably change it to `cond` and use `define`.

Reply:
Filed jackfirth/resyntax#368. I'll try some experiments and see how they go.